### ISSUE TITLE
loadexternal.sh

### ISSUE THREAD

**Issue by bongitbond** [tpm2-software/tpm2-tools#1359]
I am running the loadexternal.sh script, I am on Python 2.7.
I am receiving an error when running line no. 143:
local name1=`yaml_get_kv "stdout.yaml" "name"`

the name is not defined anywhere and not in helpers.sh

Can you provide some input?

Regards

**Comment by williamcroberts** [comment]
`name` isn't the name of variable, its the variable contents. Ie you're passing the string name as the second argument to `yaml_get_kv`. You're pullung out the name variable from the YAML compliant stdout of tpm2_loadexternal.

**Comment by bongitbond** [comment]
Thanks! Python is throwing a NameError: name 'name' is not defined. I have tested manually, it works. Just that Python is throwing that error, I have tried both Python 2and Python 3.

**Comment by bongitbond** [comment]
I am using the commit cf768cdf97864a3875e07439440d23115041cb4e. 
The run_aes_test 256 test is failing at:openssl enc -in plain.enc -out plain.dec.ssl -d -K `xxd -p sym.key` -iv 0 \
-aes-$1-cfb

OpenSSL is returning the below:
Extra Arguments given
enc: Use -help for summary
-------------------------------------------

Can you please provide some inputs? 
Is there any issue with tpm2?

**Comment by bongitbond** [comment]
The test is passing for aes 128.

**Comment by williamcroberts** [comment]
I have no idea why it would pass for 128 and fail for 256 unless your openssl doesn't support AES 256 CBC, can you check if thats supported?

**Comment by joholl** [comment]
I assume the problem is line wrapping for `xxd`, in [this line](https://github.com/tpm2-software/tpm2-tools/blob/cf768cdf97864a3875e07439440d23115041cb4e/test/integration/tests/loadexternal.sh#L148).

```c
openssl enc -in plain.enc -out plain.dec.ssl -d -K `xxd -p sym.key` -iv 0 \
-aes-$1-cfb
```

The length of `sym.key` is `16` or `32` depending on the aes used. However, `xxd` performs a line wrapping at some point which can be prevented by passing `-c 256`. The problem was fixed in 9969d48f.
